File: launchdarkly/java/src/main/java/oseg/launchdarkly_examples/PostWorkflowExample.java
package oseg.launchdarkly_examples;

import com.launchdarkly.client.ApiException;
import com.launchdarkly.client.Configuration;
import com.launchdarkly.client.api.*;
import com.launchdarkly.client.auth.*;
import com.launchdarkly.client.JSON;
import com.launchdarkly.client.model.*;

import java.io.File;
import java.time.LocalDate;
import java.time.OffsetDateTime;
import java.util.ArrayList;
import java.util.List;
import java.util.Map;

public class PostWorkflowExample
{
    public static void main(String[] args)
    {
        var config = Configuration.getDefaultApiClient();
        config.setApiKey("YOUR_API_KEY");

        var stages1Action = new ActionInput();

        var stages1Conditions1 = new ConditionInput();
        stages1Conditions1.scheduleKind("relative");
        stages1Conditions1.waitDuration(2);
        stages1Conditions1.waitDurationUnit("calendarDay");
        stages1Conditions1.kind("schedule");

        var stages1Conditions = new ArrayList<ConditionInput>(List.of (
            stages1Conditions1
        ));

        var stages1 = new StageInput();
        stages1.name("10% rollout on day 1");
        stages1.action(stages1Action);
        stages1.conditions(stages1Conditions);

        var stages = new ArrayList<StageInput>(List.of (
            stages1
        ));

        var customWorkflowInput = new CustomWorkflowInput();
        customWorkflowInput.name("Progressive rollout starting in two days");
        customWorkflowInput.description("Turn flag on for 10% of customers each day");
        customWorkflowInput.stages(stages);

        try
        {
            var response = new WorkflowsApi(config).postWorkflow(
                null, // projectKey
                null, // featureFlagKey
                null, // environmentKey
                customWorkflowInput,
                null, // templateKey
                null // dryRun
            );

            System.out.println(response);
        } catch (ApiException e) {
            System.err.println("Exception when calling WorkflowsApi#postWorkflow");
            System.err.println("Status code: " + e.getCode());
            System.err.println("Reason: " + e.getResponseBody());
            System.err.println("Response headers: " + e.getResponseHeaders());
            e.printStackTrace();
        }
    }
}

File: launchdarkly/kotlin/src/main/kotlin/oseg/launchdarkly_examples/PostWorkflowExample.kt
package oseg.launchdarkly_examples

import com.launchdarkly.client.infrastructure.*
import com.launchdarkly.client.apis.*
import com.launchdarkly.client.models.*

import java.io.File
import java.time.LocalDate
import java.time.OffsetDateTime
import kotlin.collections.ArrayList
import kotlin.collections.List
import kotlin.collections.Map
import com.squareup.moshi.adapter

@ExperimentalStdlibApi
class PostWorkflowExample
{
    fun postWorkflow()
    {
        ApiClient.apiKey["Authorization"] = "YOUR_API_KEY"

        val stages1Action = ActionInput()

        val stages1Conditions1 = ConditionInput(
            scheduleKind = "relative",
            waitDuration = 2,
            waitDurationUnit = "calendarDay",
            kind = "schedule",
        )

        val stages1Conditions = arrayListOf<ConditionInput>(
            stages1Conditions1,
        )

        val stages1 = StageInput(
            name = "10% rollout on day 1",
            action = stages1Action,
            conditions = stages1Conditions,
        )

        val stages = arrayListOf<StageInput>(
            stages1,
        )

        val customWorkflowInput = CustomWorkflowInput(
            name = "Progressive rollout starting in two days",
            description = "Turn flag on for 10% of customers each day",
            stages = stages,
        )

        try
        {
            val response = WorkflowsApi().postWorkflow(
                projectKey = "projectKey_string",
                featureFlagKey = "featureFlagKey_string",
                environmentKey = "environmentKey_string",
                customWorkflowInput = customWorkflowInput,
                templateKey = null,
                dryRun = null,
            )

            println(response)
        } catch (e: ClientException) {
            println("4xx response calling WorkflowsApi#postWorkflow")
            e.printStackTrace()
        } catch (e: ServerException) {
            println("5xx response calling WorkflowsApi#postWorkflow")
            e.printStackTrace()
        }
    }
}
